src/unitxt/inference.py

```diff
@@ -579,10 +626,14 @@ def process_data_before_dump(self, data):
 
     @staticmethod
     def _read_wml_credentials_from_env() -> (
-        Dict[Literal["url", "apikey", "project_id"], str]
+        Dict[Literal["url", "apikey", "project_id", "space_id"], str]
     ):
         credentials = {}
-        for env_var_name in ["WML_URL", "WML_PROJECT_ID", "WML_APIKEY"]:
+        project_or_deployment_var_name = (
+            "WML_SPACE_ID" if "WML_SPACE_ID" in os.environ else "WML_PROJECT_ID"
+        )
```

Comment:
what is the difference between the different env vars?

Reply:
The wml credentials can be associated with a project (id set in env var WML_PROJECT_ID) or a space (id set in env var WML_SPACE_ID), we were asked to support both options.